src/Domain/UrlExtensionExtractor.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\ExternalContent\Domain;
+
+class UrlExtensionExtractor {
+
+	public function extractExtension( string $url ): string {
+		return pathinfo(
+			parse_url( $url, PHP_URL_PATH ),
+			PATHINFO_EXTENSION
+		);
+	}
```

Comment:
I'm failing to spot the difference, but if the tests are happy...

Reply:
My old code was without `parse_url`. The code in FileExtensionUrlValidator.php has it, so I just extracted it and used it in both places.